exp/views/responses.py

```diff
@@ -1109,6 +1165,78 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class StudyResponsesFrameDataPsychDS(ResponseDownloadMixin, generic.list.ListView):
+    """Hitting this URL downloads a ZIP file in Psych-DS formatting with frame data from one response per file in CSV format"""
+
+    # TODO: with large files / many responses generation can take a while. Should generate asynchronously along
+    # with the data dict.
+    def render_to_response(self, context, **response_kwargs):
+        paginator = context["paginator"]
+        study = self.study
+
+        if study.study_type.is_external:
+            messages.error(
+                self.request, "Frame data is not available for External Studies."
+            )
+
+            return study_responses_all(study)
+
+        # build dict with contextual infomation about the study/lab/user
+        metadata_json = build_metadata_object(study)
+        # variables from column headers will be collected while looping through files
+        variables_measured = []
+
+        zipped_file = io.BytesIO()
+        with zipfile.ZipFile(zipped_file, "w", zipfile.ZIP_DEFLATED) as zipped:
+            for page_num in paginator.page_range:
+                page_of_responses = paginator.page(page_num)
+                for resp in page_of_responses:
+                    data = build_single_response_framedata_csv(resp)
+                    # get column headers from first row
+                    variables_measured += [
+                        column.strip('"')
+                        for column in data.split("\n")[0].strip().split(",")
+                    ]
+                    # psych-DS files use "keyword" formatting
+                    keywords = {
+                        "study": study_name_for_files(study.name),
+                        "response": resp.id,
+                        "child": resp.child_id,
+                    }
+                    filename = keyword_filename(keywords, "csv")
+                    zipped.writestr(f"/data/{filename}", data)
+                    # make a response-specific metadata file
+                    sidecar_metadata = {
+                        "response_uuid": str(resp.uuid),
+                        "eligibility": resp.eligibility[0],
```

Comment:
I managed to break this by trying to download responses from an older study/responses that were created before the 'eligibility' field was added to the Response objects. Since this field was added, all Response objects should have at least one value in the 'eligibility' array. However, responses that were collected _before_ this field was added will have empty arrays, and therefore produce a 'list index out of range' error on this line:
```python
"eligibility": resp.eligibility[0],
```
Also, it's possible for a response to have multiple 'ineligible' categories, e.g. 'Ineligible_TooYoung'/'Ineligible_TooOld' and 'Ineligible_CriteriaExpression'. Is it possible for the metadata eligibility value to be an array, to capture those cases?

Reply:
I checked your new version with my old data set, which contains responses with (1) missing eligibility values and (2) multiple ineligible categories. Works perfectly in both cases!